manim/utils/rate_functions.py

```diff
@@ -238,23 +243,24 @@ def not_quite_there(
     func: typing.Callable[[float], float] = smooth,
     proportion: float = 0.7,
 ) -> typing.Callable[[float], float]:
-    def result(t):
+    def result(t: float) -> float:
         return proportion * func(t)
 
     return result
 
 
 @zero
 def wiggle(t: float, wiggles: float = 2) -> float:
-    return there_and_back(t) * np.sin(wiggles * np.pi * t)
+    val: float = np.sin(wiggles * np.pi * t)
+    return there_and_back(t) * val
 
 
 def squish_rate_func(
     func: typing.Callable[[float], float],
     a: float = 0.4,
     b: float = 0.6,
 ) -> typing.Callable[[float], float]:
-    def result(t):
+    def result(t: float) -> float:
```

Comment:
```suggestion
    def result(t: float, *args: P.args, **kwargs: P.kwargs) -> float:
```

Plus, the returns could be `func(0, *args, **kwargs)`, `func(1, *args, **kwargs)` and `func((t - a) / (b - a), *args, **kwargs)`.

Reply:
I have added RateFunction[P] as the return type of squish_rate_func.